com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/MyTextDocumentServiceTest.java

```diff
@@ -254,6 +255,24 @@ public void testCodeActionsEndpoint() {
     verify(actions).collect(params);
   }
 
+  /**
+   * This test verify that when a {@link MyTextDocumentService#didClose(DidCloseTextDocumentParams)}
+   * is sent from the client to dispose a document, all the related diagnostic message are disposed
+   * from the document.
+   */
+  @Test
+  public void testDidCloseDisposeDiagnostics() {
+    service.didOpen(
+        new DidOpenTextDocumentParams(
+            new TextDocumentItem(DOCUMENT_URI, LANGUAGE, 1, TEXT_EXAMPLE)));
+    assertEquals(1, closeGetter(service).size());
+    TextDocumentIdentifier testDocument = new TextDocumentIdentifier(DOCUMENT_URI);
+    DidCloseTextDocumentParams closedDocument = new DidCloseTextDocumentParams(testDocument);
+    service.didClose(closedDocument);
+    UseCaseUtils.waitForDiagnostics(client);
```

Comment:
Could you, please, use `Communications` mock to check the logic?
The `UseCaseUtils#waitFodDiagnostic` is going to be deprecated.